packages/dx-react-grid-demos/src/demo-sources/grid-basic/basic.jsxt

```diff
@@ -9,35 +9,27 @@ import {
   globalSalesValues,
 } from '../../../demo-data/generator';
 
-export default class Demo extends React.PureComponent {
-  constructor(props) {
-    super(props);
+const Demo = () => {
+  const [columns] = useState([
+    { name: 'region', title: 'Region' },
+    { name: 'sector', title: 'Sector' },
+    { name: 'customer', title: 'Customer' },
+    { name: 'product', title: 'Product' },
+    { name: 'amount', title: 'Sale Amount' },
+  ]);
+  const [rows] = useState(generateRows({ columnValues: globalSalesValues, length: 8 }));
```

Comment:
Should we call `useState` function inside `Demo` component? It is not the same as the previous case. Inside `constructor` state calculating once, but in this case, it would be calculated every time on it rendered.

Reply:
We doesn't call `useState` every render. It is called ones, on first render and this variables are preserved by React.
https://reactjs.org/docs/hooks-state.html#declaring-a-state-variable